src/select/Select.ts

```diff
@@ -18,382 +17,332 @@ import * as iconCss from '../common/styles/icons.m.css';
- * @property value          The current value
- * @property onBlur         Called when the input loses focus
- * @property onChange       Called when the node's 'change' event is fired
- * @property onClick        Called when the input is clicked
- * @property onFocus        Called when the input is focused
- * @property onKeyDown      Called on the input's keydown event
+ * @property describedBy       ID of an element that provides more descriptive text
+ * @property disabled          Prevents the user from interacting with the form field
+ * @property invalid           Indicates the value entered in the form field is invalid
+ * @property getOptionDisabled Function that accepts an option's data and index and returns a boolean
+ * @property getOptionId       Function that accepts an option's data and index and returns a string id
+ * @property getOptionLabel    Function that accepts an option's data and index and returns a DNode label
+ * @property getOptionSelected Function that accepts an option's data and index and returns a boolean
+ * @property getOptionValue    Function that accepts an option's data and index and returns a string value
+ * @property label             Label settings for form label text, position, and visibility
+ * @property name              The form widget's name
+ * @property options           Array of any type of data for the options
+ * @property readOnly          Allows or prevents user interaction
+ * @property required          Whether or not a value is required
+ * @property useNativeElement  Use the native <select> element if true
+ * @property value             The current value
+ * @property onBlur            Called when the input loses focus
+ * @property onChange          Called when the node's 'change' event is fired
+ * @property onFocus           Called when the input is focused
  */
 export interface SelectProperties extends ThemeableProperties {
-	CustomOption?: Constructor<SelectOption>;
 	describedBy?: string;
 	disabled?: boolean;
 	invalid?: boolean;
+	getOptionDisabled?(option: any, index: number): boolean;
+	getOptionId?(option: any, index: number): string;
+	getOptionLabel?(option: any): DNode;
+	getOptionSelected?(option: any, index: number): boolean;
+	getOptionValue?(option: any, index: number): string;
 	label?: string | LabelOptions;
-	multiple?: boolean;
 	name?: string;
-	options?: OptionData[];
+	options?: any[];
 	readOnly?: boolean;
 	required?: boolean;
 	useNativeElement?: boolean;
 	value?: string;
-	onBlur?(event: FocusEvent): void;
-	onChange?(option: OptionData): void;
-	onClick?(event: MouseEvent): void;
-	onFocus?(event: FocusEvent): void;
-	onKeyDown?(event: KeyboardEvent): void;
+	onBlur?(key: string | number): void;
+	onChange?(option: any, key: string | number): void;
+	onFocus?(key: string | number): void;
 }
 
 export const SelectBase = ThemeableMixin(WidgetBase);
 
 @theme(css)
 @theme(iconCss)
+@diffProperty('options', reference)
```

Comment:
Was this working incorrectly before explicitly diffing by reference?

Reply:
Yup, it doesn't call a render if you update the `options` property otherwise.